README.md

```diff
@@ -1 +1,293 @@
+ So with simple migrations you shouldn't worry about the difference between `datastore-fs` and `datastore-level` 
+ and by default use the `datastore-fs` package (as the replace mechanism does not work vice versa).
+
+#### Guidelines
+
+The recommended way to write a new migration is to first bootstrap a dummy migration using the CLI:
+
+```sh
+> jsipfs-migrations add
+```
+
+A new folder is created with the bootstrapped migration. You can then simply fill in the required fields and 
+write the rest of the migration! 
+
+#### Migration's dependencies
+
+The size of the `js-ipfs` bundle is crucial for distribution in a browser environment, so dependency management of all related
+packages is important.
+
+If a migration needs to depend on some package, this dependency should be declared in the root's `package.json`. The author
+of the migration should be thoughtful about adding dependencies that would significantly increase the size of the final bundle.
+
+Most of the migration's dependencies will most likely overlap with `js-ipfs`'s dependencies and hence should not introduce
+any significant overhead, but it is necessary to keep the versions of these dependencies in sync with `js-ipfs`. For this 
+reason migrations should be well tested to ensure correct behaviour over dependency updates.
+An update of some dependency could introduce breaking change. In such a case the next steps should be discussed with a broader 
+audience. 
+
+#### Integration with js-ipfs
+
+When a new migration is created, the repo version in [`js-ipfs-repo`](https://github.com/ipfs/js-ipfs-repo) should be updated with the new version,
+together with updated version of this package. Then the updated version should be propagated to `js-ipfs`.
+
+#### Tests
+
+If a migration affects any of the following functionality, it must provide tests for the following functions
+ to work under the version of the repo that it migrates to:
+
+* `/src/repo/version.js`:`getVersion()` - retrieving repository's version
+* `/src/repo/lock.js`:`lock()` - locking repository that uses file system
+* `/src/repo/lock-memory.js`:`lock()` - locking repository that uses memory
+
+Every migration must have test coverage. Tests for migrations should be placed in the `/test/migrations/` folder. Most probably
+you will have to plug the tests into `browser.js`/`node.js` if they require specific bootstrapping on each platform.
+
+#### Empty migrations
+
+For interop with go-ipfs it might be necessary just to bump a version of a repo without any actual 
+modification as there might not be any changes needed in the JS implementation. For that purpose you can create an "empty migration".
+
+The easiest way to do so is with the CLI:
+
+```sh
+> jsipfs-migrations add --empty
+```
+
+This will create an empty migration with the next version.
+
+### Migrations matrix
+
+| IPFS repo version  | JS IPFS version  |
+| -----------------: |:----------------:|
+|                  7 | v0.0.0 - latest  |
+
+
+## API
```

Comment:
The API of the module should come before the explanation of how to create a new migration as that is how most people will use the module